pydantic/_internal/_decorators.py

```diff
@@ -383,7 +385,9 @@ def gather_decorator_functions(cls: type[Any]) -> DecoratorInfos:
 _FUNCS: set[str] = set()
 
 
-AnyDecoratorCallable: TypeAlias = 'Union[classmethod[Any], staticmethod[Any], partialmethod[Any], Callable[..., Any]]'
+AnyDecoratorCallable: TypeAlias = (
+    'Union[classmethod[Any, Any, Any], staticmethod[Any, Any], partialmethod[Any], Callable[..., Any]]'
```

Comment:
we should confirm this doesn't break mypy for users, or at least be prepared to accept the fact that older mypy won't work. Maybe we can use the `if MYPY` trick, I think there's a way to do things differently based on mypy version info too (like sys.version_info)

Reply:
We run mypy in tests, that kinda confirms it right?

Reply:
Assuming one of the mypy tests covers this. I think it's probably fine, but I guess we can deal with it if/when people report issues. (And can tell them to upgrade mypy I guess lol.)